### PR TITLE
Add support for Login and User management

### PR DESCRIPTION
**Pull request by alanrenmsft**
Send this out for review while waiting for another STS update with some bug fixes.

New Login
new-login

Update Login
update-login

Delete Login
delete-login

New User
new-user

Update User
update-user

Delete User
image

### CODE REVIEW

File: extensions/mssql/package.nls.json [hunk 1]
```diff
@@ -179,8 +179,14 @@
 	"title.designTable": "Design",
 	"mssql.parallelMessageProcessing" : "[Experimental] Whether the requests to the SQL Tools Service should be handled in parallel. This is introduced to discover the issues there might be when handling all requests in parallel. The default value is false. Relaunch of ADS is required when the value is changed.",
 	"mssql.tableDesigner.preloadDatabaseModel": "Whether to preload the database model when the database node in the object explorer is expanded. When enabled, the loading time of table designer can be reduced. Note: You might see higher than normal memory usage if you need to expand a lot of database nodes.",
-
 	"mssql.objectExplorer.groupBySchema": "When enabled, the database objects in Object Explorer will be categorized by schema.",
 	"mssql.objectExplorer.enableGroupBySchema":"Enable Group By Schema",
-	"mssql.objectExplorer.disableGroupBySchema":"Disable Group By Schema"
+	"mssql.objectExplorer.disableGroupBySchema":"Disable Group By Schema",
+	"title.newServerRole": "New Server Role",
+	"title.newLogin": "New Login",
```
Comment (kburtram) [inline extensions/mssql/package.nls.json:186]: can we add the string "Preview" to these commands for now?

**Review by kburtram — APPROVED** [review]
LGTM as an initial WIP. We can continue to enhance incrementally after getting the initial feature structure in place.

File: extensions/mssql/package.nls.json (hunk 1 above)
Comment (alanrenmsft) [reply extensions/mssql/package.nls.json:186]: I've already added Preview to the dialog title.
image

File: extensions/mssql/package.json [hunk 2]
```diff
@@ -404,6 +452,26 @@
...
+          "when": "connectionProvider == MSSQL && nodeType == Folder && objectType == ServerLevelLogins && config.workbench.enablePreviewFeatures",
+          "group": "0_query@1"
+        },
+        {
+          "command": "mssql.newUser",
+          "when": "connectionProvider == MSSQL && nodeType == Folder && objectType == Users && config.workbench.enablePreviewFeatures",
+          "group": "0_query@1"
+        },
+        {
+          "command": "mssql.objectProperties",
```
Comment (Charles-Gagnon) [inline extensions/mssql/package.json:466]: We'll probably want to make the Admin Tools extension not show a duplicate properties menu item - although given how much more complex the UI is there maybe we should keep it but just rename it to make it clear it's not the "default" properties item

File: extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	}
+
+	protected get originalObjectInfo(): ObjectInfoType {
+		return this._originalObjectInfo;
+	}
+
+	public async open(): Promise<void> {
+		await vscode.window.withProgress({
+			location: vscode.ProgressLocation.Notification,
+			title: LoadingDialogText
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts:104]: Not a huge fan of the toast being used for this - is there a reason you're not showing the dialog immediately and then just having a loading spinner while it loads? We do that pretty much everywhere else.

File: extensions/mssql/package.json (hunk 2 above)
Comment (alanrenmsft) [reply extensions/mssql/package.json:466]: I thought about it and reverted my changes in the ssms-min extension, until we have completed all the features in Login/User, I will keep both present, I might be a little bit confusing but user should be able to figure out the differences easily, plus the menu item group is different.

File: extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts (hunk 3 above)
Comment (alanrenmsft) [reply extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts:104]: I am planning to do it in a follow up PR.

**Pull request merged**